### PR TITLE
Support AWS sms feature and fix phonenumber verification

### LINKED ISSUE #140: if AWS is supported as the provider for SMS services
I saw in the notification-manager documentation that the provider supports huawei, aliyun, tencent when the receiver is SMS. I would like to know if AWS is currently supported as the provider for SMS services.

https://github.com/kubesphere/notification-manager/blob/master/docs/crds/receiver.md#sms-receiver

### LINKED ISSUE #155: sms receiver: 17212345678 is not a valid phone number
I use api `/api/v2/verify` to verify the SMS, but I always get the following error, no matter what`phoneNumbers` is.
```
{"Status":400,"Message":"sms receiver: 17212345678 is not a valid phone number"}

{"Status":400,"Message":"sms receiver: +8617212345678 is not a valid phone number"}
```

### PR DESCRIPTION
**Pull request by Bennu-Li**
Signed-off-by: Bennu-Li <yunmei.li@zilliz.com>

fixed #140 
fixed #155

### CODE REVIEW

File: pkg/apis/v2beta2/config_webhook.go [hunk 1]
```diff
@@ -168,6 +169,22 @@ func (r *Config) validateConfig() error {
 			}
 		}
 
+		// Sms AWS provider parameters validation
+		if providers.AWS != nil {
+			if providers.AWS.AccessKeyId != nil {
```
Comment (wanjunlei) [inline pkg/apis/v2beta2/config_webhook.go:174]: The `AccessKeyId` and `SecretAccessKey` can be nil?

File: pkg/notify/notifier/sms/aws.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	accessKeyId, err := a.notifierCtl.GetCredential(a.AccessKeyId)
+	if err != nil {
+		return utils.Errorf("[AWS  SendSms] cannot get accessKeyId: %s", err.Error())
+	}
+	secretAccessKey, err := a.notifierCtl.GetCredential(a.SecretAccessKey)
+	if err != nil {
+		return utils.Errorf("[AWS  SendSms] cannot get secretAccessKey: %s", err.Error())
+	}
+
+	err = os.Setenv("AWS_ACCESS_KEY_ID", accessKeyId)
```
Comment (wanjunlei) [inline pkg/notify/notifier/sms/aws.go:58]: If there are two configs with different accessKeyId, this environment variable will be overwritten, is this as expected?

File: pkg/apis/v2beta2/config_webhook.go (hunk 1 above)
Comment (Bennu-Li) [reply pkg/apis/v2beta2/config_webhook.go:174]: No, AccessKeyId and SecretAccessKey can't be nil.

File: pkg/notify/notifier/sms/aws.go (hunk 2 above)
Comment (wanjunlei) [reply pkg/notify/notifier/sms/aws.go:58]: If there is a receiver `a` with config `a` like this.

```
spec:
  sms:
    aws:
      accessKeyId: 
        value: accessKeyId1
      secretAccessKey:
        value: secretAccessKey1
```

and a receiver `b` with config `b` like this.

```
spec:
  sms:
    aws:
      accessKeyId: 
        value: accessKeyId2
      secretAccessKey:
        value: secretAccessKey2
```

If these two receivers send notifications at the same time, the `AWS_ACCESS_KEY_ID` will be set twice? What is the value of `AWS_ACCESS_KEY_ID` when sending the short message, `accessKeyId1` or `accessKeyId2`?

Comment (Bennu-Li) [reply pkg/notify/notifier/sms/aws.go:58]: Good question, I'll try to resolve it.

File: pkg/notify/notifier/sms/aws.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	if err != nil {
+		return utils.Errorf("[AWS  SendSms] set environment variables AWS_ACCESS_KEY_ID: %s", err.Error())
+	}
+	err = os.Setenv("AWS_SECRET_ACCESS_KEY", secretAccessKey)
+	if err != nil {
+		return utils.Errorf("[AWS  SendSms] set environment variables AWS_SECRET_ACCESS_KEY: %s", err.Error())
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(a.Region))
```
Comment (zhu733756) [inline pkg/notify/notifier/sms/aws.go:67]: We just need one context here，pls use “ctx = context.TODO()” as the reference.

See https://github.com/kubesphere/notification-manager/blob/master/pkg/notify/notifier/sms/sms.go#L104

File: pkg/notify/notifier/sms/aws.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	}
+
+	client := sns.NewFromConfig(cfg)
+
+	input := &sns.PublishInput{
+		Message:     &messages,
+		PhoneNumber: &a.PhoneNums,
+	}
+
+	_, err = PublishMessage(context.TODO(), client, input)
```
Comment (wanjunlei) [inline pkg/notify/notifier/sms/aws.go:77]: It should use the context passed by the caller, not `context. TODO()`, line 60 has the same problem.

**Review by wanjunlei — APPROVED** [review]

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
Thank you @Bennu-Li for the PR

**Pull request merged**